ruby: linker errors

gem version 1.3.1 installs fine. 1.3.2 and current master fail linking:

```
linking shared-object whisper.bundle
ld: warning: object file (libggml.a(ggml-backend-reg.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libwhisper.a(whisper.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-blas.a(ggml-blas.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-metal.a(ggml-metal.m.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(ggml-cpu.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libcommon.a(common.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(ggml-cpu.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(repack.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(hbm.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(quants.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libcommon.a(common-ggml.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(traits.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(amx.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(mmq.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(binary-ops.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml-alloc.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libcommon.a(common-whisper.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(unary-ops.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml-backend.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-metal.a(ggml-metal-embed.s.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(vec.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml-opt.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(ops.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml-threading.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(ggml-quants.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(quants.c.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-base.a(gguf.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libggml-cpu.a(repack.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
ld: warning: object file (libcommon.a(grammar-parser.cpp.o)) was built for newer macOS version (12.7) than being linked (12.0)
Undefined symbols for architecture arm64:
  "_MTLCopyAllDevices", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
  "_MTLCreateSystemDefaultDevice", referenced from:
      _ggml_backend_metal_device_acq in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_MTLCaptureDescriptor", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_MTLCaptureManager", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_MTLCompileOptions", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_MTLHeapDescriptor", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSMutableArray", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSMutableDictionary", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSNumber", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSObject", referenced from:
      _OBJC_CLASS_$_ggml_metal_heap_ptr in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSString", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_CLASS_$_NSURL", referenced from:
      objc-class-ref in libggml-metal.a(ggml-metal.m.o)
  "_OBJC_METACLASS_$_NSObject", referenced from:
      _OBJC_METACLASS_$_ggml_metal_heap_ptr in libggml-metal.a(ggml-metal.m.o)
  "___CFConstantStringClassReference", referenced from:
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      CFString in libggml-metal.a(ggml-metal.m.o)
      ...
  "__objc_empty_cache", referenced from:
      _OBJC_METACLASS_$_ggml_metal_heap_ptr in libggml-metal.a(ggml-metal.m.o)
      _OBJC_CLASS_$_ggml_metal_heap_ptr in libggml-metal.a(ggml-metal.m.o)
  "_cblas_sgemm", referenced from:
      ggml_backend_blas_graph_compute(ggml_backend*, ggml_cgraph*) in libggml-blas.a(ggml-blas.cpp.o)
  "_objc_alloc", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
  "_objc_alloc_init", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_mem_pool_alloc in libggml-metal.a(ggml-metal.m.o)
  "_objc_autoreleasePoolPop", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_graph_compute in libggml-metal.a(ggml-metal.m.o)
  "_objc_autoreleasePoolPush", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_graph_compute in libggml-metal.a(ggml-metal.m.o)
  "_objc_enumerationMutation", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_free in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_heap_reset in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_encode_node in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_mem_pool_alloc in libggml-metal.a(ggml-metal.m.o)
  "_objc_opt_new", referenced from:
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_graph_compute in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_mem_pool_alloc in libggml-metal.a(ggml-metal.m.o)
  "_objc_release", referenced from:
      _ggml_backend_metal_buffer_type_alloc_buffer in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_buffer_type_get_max_size in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_init in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_buffer_free_buffer in libggml-metal.a(ggml-metal.m.o)
      _ggml_backend_metal_free in libggml-metal.a(ggml-metal.m.o)
      _ggml_metal_heap_reset in libggml-metal.a(ggml-metal.m.o)
      ___ggml_backend_metal_set_n_cb_block_invoke in libggml-metal.a(ggml-metal.m.o)
      ...
  "_vDSP_maxv", referenced from:
      _ggml_compute_forward_soft_max in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_flash_attn_back in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_cross_entropy_loss in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_cross_entropy_loss_back in libggml-cpu.a(ops.cpp.o)
  "_vDSP_sve", referenced from:
      _ggml_compute_forward_sum_rows in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_mean in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_cross_entropy_loss in libggml-cpu.a(ops.cpp.o)
  "_vDSP_vadd", referenced from:
      _ggml_compute_forward_add1 in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_acc in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_add_non_quantized in libggml-cpu.a(binary-ops.cpp.o)
  "_vDSP_vdiv", referenced from:
      _ggml_compute_forward_div in libggml-cpu.a(binary-ops.cpp.o)
  "_vDSP_vmul", referenced from:
      _ggml_compute_forward_mul in libggml-cpu.a(binary-ops.cpp.o)
  "_vDSP_vsmul", referenced from:
      _ggml_compute_forward_norm in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_rms_norm in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_rms_norm_back in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_group_norm in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_l2_norm in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_scale in libggml-cpu.a(ops.cpp.o)
      _ggml_compute_forward_soft_max in libggml-cpu.a(ops.cpp.o)
      ...
  "_vDSP_vsub", referenced from:
      _ggml_compute_forward_sub in libggml-cpu.a(binary-ops.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

i dug around the source, makefiles, couldn't make out anything valuable. looks like it's related to #1344.

```
$ ruby -v
ruby 3.4.4 (2025-05-14 revision a38531fd3f) +PRISM [arm64-darwin21]

$ sw_vers
ProductName:	macOS
ProductVersion:	12.7.6
BuildVersion:	21H1320
```

--- a/bindings/ruby/.gitignore
+++ b/bindings/ruby/.gitignore
@@ -6,3 +6,4 @@ ext/ggml/
 ext/include/
 ext/scripts/
 ext/src/
+test/fixtures/

--- a/bindings/ruby/Rakefile
+++ b/bindings/ruby/Rakefile
@@ -69,6 +69,21 @@ CLEAN.include LIB_FILE
 
 Rake::TestTask.new
 
+TEST_FIXTURE_AUDIO = "test/fixtures/jfk.wav"
+TEST_FIXTURE_AUDIO_SRC = File.expand_path(File.join(__dir__, "..", "..", "samples", "jfk.wav"))
+TEST_FIXTURE_AUDIO_DIR = TEST_FIXTURE_AUDIO.pathmap("%d")
+directory TEST_FIXTURE_AUDIO_DIR
+if File.exist? TEST_FIXTURE_AUDIO_SRC
+  file TEST_FIXTURE_AUDIO => [TEST_FIXTURE_AUDIO_SRC, TEST_FIXTURE_AUDIO_DIR] do |t|
+    symlink t.source, t.name
+  end
+else
+  require "open-uri"
+  file TEST_FIXTURE_AUDIO => TEST_FIXTURE_AUDIO_DIR do |t|
+    File.write t.name, URI("https://github.com/ggml-org/whisper.cpp/raw/refs/heads/master/samples/jfk.wav").read
+  end
+end
+
 TEST_MEMORY_VIEW = "test/jfk_reader/jfk_reader.#{RbConfig::CONFIG['DLEXT']}"
 file TEST_MEMORY_VIEW => "test/jfk_reader/jfk_reader.c" do |t|
   chdir "test/jfk_reader" do
@@ -76,6 +91,6 @@ file TEST_MEMORY_VIEW => "test/jfk_reader/jfk_reader.c" do |t|
     sh "make"
   end
 end
-CLEAN.include "test/jfk_reader/jfk_reader.{o,#{RbConfig::CONFIG['DLEXT']}}"
+CLEAN.include TEST_MEMORY_VIEW
 
-task test: [LIB_FILE, TEST_MEMORY_VIEW]
+task test: [LIB_FILE, TEST_MEMORY_VIEW, TEST_FIXTURE_AUDIO]

--- a/bindings/ruby/ext/options.rb
+++ b/bindings/ruby/ext/options.rb
@@ -50,7 +50,9 @@ class Options
 
   # See ggml/src/ggml-cpu/CMakeLists.txt
   def configure_accelerate
-    $LDFLAGS << " -framework Accelerate" if enabled?("GGML_ACCELERATE")
+    if RUBY_PLATFORM.match?(/darwin/) && enabled?("GGML_ACCELERATE")
+      $LDFLAGS << " -framework Accelerate"
+    end
   end
 
   # See ggml/src/ggml-metal/CMakeLists.txt

--- a/bindings/ruby/ext/options.rb
+++ b/bindings/ruby/ext/options.rb
@@ -43,7 +43,29 @@ class Options
       @options[name] = [type, value]
     end
 
+    configure_accelerate
+    configure_metal
     configure_coreml
+  end
+
+  # See ggml/src/ggml-cpu/CMakeLists.txt
+  def configure_accelerate
+    use_accelerate = if @options["GGML_ACCELERATE"][1].nil?
+                       cmake_options["GGML_ACCELERATE"][1]
+                     else
+                       @options["GGML_ACCELERATE"][1]
+                     end
+    $LDFLAGS << " -framework Accelerate" if use_accelerate
+  end
+
+  # See ggml/src/ggml-metal/CMakeLists.txt
+  def configure_metal
+    use_metal = if @options["GGML_METAL"][1].nil?
+                  cmake_options["GGML_METAL"][1]
+                else
+                  @options["GGML_METAL"][1]
+                end
+    $LDFLAGS << " -framework Foundation -framework Metal -framework MetalKit" if use_metal
   end
 
   def configure_coreml

--- a/bindings/ruby/ext/options.rb
+++ b/bindings/ruby/ext/options.rb
@@ -58,8 +58,12 @@ class Options
     $LDFLAGS << " -framework Foundation -framework Metal -framework MetalKit" if enabled?("GGML_METAL")
   end
 
+  # See src/CmakeLists.txt
   def configure_coreml
-    $CPPFLAGS << " -DRUBY_WHISPER_USE_COREML" if enabled?("WHISPER_COREML")
+    if enabled?("WHISPER_COREML")
+      $LDFLAGS << " -framework Foundation -framework CoreML"
+      $CPPFLAGS << " -DRUBY_WHISPER_USE_COREML"
+    end
   end
 
   def option_name(name)

--- a/bindings/ruby/ext/options.rb
+++ b/bindings/ruby/ext/options.rb
@@ -50,34 +50,27 @@ class Options
 
   # See ggml/src/ggml-cpu/CMakeLists.txt
   def configure_accelerate
-    use_accelerate = if @options["GGML_ACCELERATE"][1].nil?
-                       cmake_options["GGML_ACCELERATE"][1]
-                     else
-                       @options["GGML_ACCELERATE"][1]
-                     end
-    $LDFLAGS << " -framework Accelerate" if use_accelerate
+    $LDFLAGS << " -framework Accelerate" if enabled?("GGML_ACCELERATE")
   end
 
   # See ggml/src/ggml-metal/CMakeLists.txt
   def configure_metal
-    use_metal = if @options["GGML_METAL"][1].nil?
-                  cmake_options["GGML_METAL"][1]
-                else
-                  @options["GGML_METAL"][1]
-                end
-    $LDFLAGS << " -framework Foundation -framework Metal -framework MetalKit" if use_metal
+    $LDFLAGS << " -framework Foundation -framework Metal -framework MetalKit" if enabled?("GGML_METAL")
   end
 
   def configure_coreml
-    use_coreml = if @options["WHISPER_COREML"][1].nil?
-                   cmake_options["WHISPER_COREML"][1]
-                 else
-                   @options["WHISPER_COREML"][1]
-                 end
-    $CPPFLAGS << " -DRUBY_WHISPER_USE_COREML" if use_coreml
+    $CPPFLAGS << " -DRUBY_WHISPER_USE_COREML" if enabled?("WHISPER_COREML")
   end
 
   def option_name(name)
     name.downcase.gsub("_", "-")
+  end
+
+  def enabled?(option)
+    if @options[option][1].nil?
+      cmake_options[option][1]
+    else
+      @options[option][1]
+    end
   end
 end

--- a/bindings/ruby/lib/whisper/model/uri.rb
+++ b/bindings/ruby/lib/whisper/model/uri.rb
@@ -132,13 +132,13 @@ module Whisper
 
     class ZipURI < URI
       def cache
-        zip_path = Pathname(super)
+        zip_path = super
         dest = unzipped_path
         return if dest.exist? && dest.mtime >= zip_path.mtime
         escaping dest do
           system "unzip", "-q", "-d", zip_path.dirname.to_path, zip_path.to_path, exception: true
         end
-        zip_path.to_path
+        zip_path
       end
 
       def clear_cache

--- a/bindings/ruby/sig/whisper.rbs
+++ b/bindings/ruby/sig/whisper.rbs
@@ -412,7 +412,7 @@ module Whisper
     end
 
     class ZipURI < URI
-      def cache: () -> String
+      def cache: () -> Pathname
       def clear_cache: () -> void
     end
   end

--- a/bindings/ruby/test/helper.rb
+++ b/bindings/ruby/test/helper.rb
@@ -3,7 +3,7 @@ require "whisper"
 require_relative "jfk_reader/jfk_reader"
 
 class TestBase < Test::Unit::TestCase
-  AUDIO = File.join(__dir__, "..", "..", "..", "samples", "jfk.wav")
+  AUDIO = File.join(__dir__, "fixtures", "jfk.wav")
 
   class << self
     def whisper

--- a/bindings/ruby/whispercpp.gemspec
+++ b/bindings/ruby/whispercpp.gemspec
@@ -4,7 +4,6 @@ Gem::Specification.new do |s|
   s.name    = "whispercpp"
   s.authors = ["Georgi Gerganov", "Todd A. Fisher"]
   s.version = '1.3.3'
-  s.date    = '2025-06-10'
   s.description = %q{High-performance inference of OpenAI's Whisper automatic speech recognition (ASR) model via Ruby}
   s.email   = 'todd.fisher@gmail.com'
   s.extra_rdoc_files = ['LICENSE', 'README.md']

